plugins/services/src/js/events/__tests__/SDKEndpointsActions-test.js

```diff
@@ -0,0 +1,156 @@
+
+      it("sends data to the correct URL", function() {
+        expect(this.configuration.url).toEqual(
+          `${Config.rootUrl}/service/${serviceId}/v1/endpoints`
+        );
+      });
+
+      it("uses GET for the request method", function() {
+        expect(this.configuration.method).toEqual("GET");
+      });
+
+      it("dispatches the correct action when successful", function() {
+        const id = AppDispatcher.register(function(payload) {
+          const action = payload.action;
+          AppDispatcher.unregister(id);
+          expect(action).toEqual({
+            type: ActionTypes.REQUEST_SDK_ENDPOINTS_SUCCESS,
+            data: {
+              serviceId: "foo",
+              endpoints: ["endpoint1", "endpoint2"]
+            }
+          });
+        });
+
+        this.configuration.success(["endpoint1", "endpoint2"]);
+      });
+
+      it("dispatches the correct action when unsuccessful", function() {
+        var id = AppDispatcher.register(function(payload) {
+          var action = payload.action;
+          AppDispatcher.unregister(id);
+          expect(action).toEqual({
+            data: { error: {}, serviceId: "foo" },
+            type: "REQUEST_SDK_ENDPOINTS_ERROR",
+            xhr: { error: {}, serviceId: "foo" }
+          });
+        });
+
+        this.configuration.error({ error: {}, serviceId: "foo" });
+      });
+
+      it("dispatches the xhr when unsuccessful", function() {
+        var id = AppDispatcher.register(function(payload) {
+          var action = payload.action;
+          AppDispatcher.unregister(id);
+          expect(action.xhr).toEqual({
+            responseJSON: { description: "foo" }
+          });
+        });
+
+        this.configuration.error({
+          responseJSON: { description: "foo" }
+        });
+      });
+    });
+  });
+
+  describe("#fetchEndpoint", function() {
+    let mockXhr;
+    const originalXhr = global.XMLHttpRequest;
+    const serviceId = "foo";
+    const endpointName = "arangodb";
+
+    beforeEach(function() {
+      mockXhr = {
+        open: jest.fn(),
+        send: jest.fn(),
+        readyState: 4,
+        status: 200,
+        getResponseHeader: () => "application/json",
+        response: JSON.stringify({
+          data: "some data"
+        })
+      };
+
+      global.XMLHttpRequest = jest.fn(() => mockXhr);
+    });
+
+    afterEach(function() {
+      global.XMLHttpRequest = originalXhr;
+    });
+
+    it("XMLHttpRequest open was called", function() {
+      SDKEndpointsActions.fetchEndpoint(serviceId, endpointName);
+
+      expect(mockXhr.open).toHaveBeenCalled();
+    });
+
+    it("call open with GET request method and correct url", function() {
+      const url = `/service/${serviceId}/v1/endpoints/${endpointName}`;
+
+      SDKEndpointsActions.fetchEndpoint(serviceId, endpointName);
+
+      expect(mockXhr.open).toHaveBeenCalledWith("GET", url);
+    });
+
+    it("dispatches the correct action when successful", function() {
```

Comment:
Add another test case for properly handling a response that is *not* JSON?

Reply:
Included a test case for text.